apps/notifications/package.json

```diff
@@ -18,10 +18,11 @@
 	"homepage": "https://github.com/Automattic/wp-calypso",
 	"scripts": {
 		"clean": "npx rimraf dist",
-		"build": "yarn clean && BROWSERSLIST_ENV=wpcom calypso-build --config='./webpack.config.js'",
+		"build:notifications": "calypso-build --config='./webpack.config.js'",
```

Comment:
Could you elaborate on those namespaced commands? I see you referenced them as additional entrypoints, but they weren't defined before. Why are those needed and why are they here now?

Reply:
Also, I find it a bit confusing that `build:notifications` calls `calypso-build`, while `build` calls `dev`. Could you elaborate here, too?

Reply:
Great question! Here's how it works:

So `calypso-build` is the root build task. It will call webpack and do the actual work to build the plugin. However, calypso build by itself doesn't (and probably shouldn't) include any code which would let us sync anything to a sandbox at the same time. In order to allow the sync operation to happen at the same time, we need to instead call our own script which can orchestrate everything.

Our own script which orchestrates everything will run all build tasks defined by the plugin in parallel, and will also start the sync part automatically when the flag is specified.

So basically:
- Both build and dev call `build-calypso-app.mjs`. (The only difference between build/dev is the NODE_ENV)
- `build-calypso-app.mjs` starts the sync script based on the presence of `--sync`
- `build-calypso-app.mjs` handles runing in watch mode or production mode based on the NODE_ENV.
- `build-calypso-app.mjs` starts all `build:*` tasks defined in package.json in parallel.
- the `build:*` tasks call calypso-build 
- `calypso-build` calls `webpack`

We _could_ have another package script which calls the build-calypso-app script, and then both build and dev call that instead. But IMO it just takes up extra space, since we can just put that call into either build or dev, and then have the other call that.

The script is important because we can't easily orchestrate all this stuff using _just_ package.json scripts (nor can we keep it consistent.) The namespacing for `build:` is important because some apps define multiple webpack builds, all of which need to run together (ideally in parallel). For example, ETK has 11 or 12 different webpack builds to handle each sub-plugin.

Notifications doesn't have multiple webpack builds, so it is a little weird here. But we need some kind of format for the actual webpack build so that our build-calyps-app script knows exactly what to call. (`npm-run-all` can handle wildcards, which lets us use `build:*` to run webpack)

I'm happy for suggestions because I know it's a bit confusing.